### ISSUE TITLE
[Firestore] CloudFirestorePlugin.java uses unchecked or unsafe operations.

### ISSUE THREAD

**Issue by kingdg25** [FirebaseExtended/flutterfire#1926]
```bash
Note: /home/brooky/flutter/.pub-cache/hosted/pub.dartlang.org/cloud_firestore-0.13.0+1/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

```

in debug mode, the app runs perfectly, but when running using the release apk(build apk), it crashes the app when opening.

**Comment by iapicca** [comment]
Hi @kingdg25 
can you please provide your `flutter doctor -v` , 
your `flutter build apk --verbose` 
and your `pubspec.yaml` 
or if possible a reproducible minimal code sample.
Thank you

**Comment by kingdg25** [comment]
my pubspec.yaml: `description: A new Flutter application.

version: 1.0.0+1

environment:
  sdk: ">=2.0.0-dev.68.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  cupertino_icons: ^0.1.2
  animated_text_kit: ^1.3.0
  firebase_core: ^0.4.1+5
  firebase_auth: ^0.11.0
  cloud_firestore: ^0.11.0
  modal_progress_hud: ^0.1.3
  http: ^0.12.0+2
  photo_view: ^0.9.0
  intl: ^0.16.0
  cached_network_image: ^2.0.0
  carousel_slider: ^1.3.1
  fluttertoast: ^3.1.3
  shared_preferences: ^0.5.3+4
  google_sign_in: ^4.0.11
  flutter_auth_buttons: ^0.6.0
  flutter_facebook_login: ^3.0.0
  esys_flutter_share: ^1.0.2
  random_string: ^2.0.1
  percent_indicator: ^2.1.1
  connectivity: ^0.4.6
  intro_slider: ^2.2.8
  provider: ^3.2.0
  dio: ^3.0.8
  path_provider: ^1.5.0`
flutter doctor --verbose
`[✓] Flutter (Channel stable, v1.12.13+hotfix.7, on Linux, locale en_US.utf8)
    • Flutter version 1.12.13+hotfix.7 at /home/brooky/flutter
    • Framework revision 9f5ff2306b (7 days ago), 2020-01-26 22:38:26 -0800
    • Engine revision a67792536c
    • Dart version 2.7.0

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /home/brooky/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /home/brooky/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Android Studio (version 3.5)
    • Android Studio at /home/brooky/android-studio
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[!] VS Code (version 1.37.0)
    • VS Code at /usr/share/code
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (1 available)
    • SM A207F • R9AM90WQLPJ • android-arm64 • Android 9 (API 28)`

**Comment by rishabhBlr** [comment]
I am also getting the same error..

android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
 
Yaml file below
version: 1.0.0+1

environment:
  sdk: ">=2.6.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  splashscreen: ^1.2.0
  datetime_picker_formfield: ^1.0.0
  cloud_firestore: ^0.13.0+1
  firebase_core: ^0.4.0

**Comment by sajjad-hk** [comment]
Similar issue here! any update?

**Comment by ouardia-y** [comment]
Any update? I'm having the same issue. I ran flutter doctor and there are no issues found.

**Comment by erwinvasquez** [comment]
any update? please!!! I have the same issue.
Note: /Users/erwinvasquezpinto/Programacion/Flutter/.pub-cache/hosted/pub.dartlang.org/cloud_firestore-0.13.5/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java uses unchecked or unsafe operations.

**Comment by shinriyo** [comment]
related #1401 ?

**Comment by Salakar** [comment]
Hey all, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
